spec/services/target/create_target_services_spec.rb

```diff
@@ -40,5 +40,61 @@
         expect { subject }.to raise_error(MaxAllowTargetsError)
       end
     end
+
+    context 'when two targets match' do
+      let!(:target) do
+        create(:target,
+               latitude: target_params[:latitude],
+               longitude: target_params[:longitude],
+               topic_id: topic_id)
+      end
+
+      it 'Target connect with other target' do
+        expect { subject }.to change { Match.count }.from(0).to(1)
+      end
+    end
+
+    context 'when two targets does not match' do
+      context 'when has different topic' do
+        let!(:target) do
+          create(:target,
+                 latitude: target_params[:latitude],
+                 longitude: target_params[:longitude])
+        end
+
+        it 'does not create a connection' do
+          expect { subject }.not_to change { Match.count }
+        end
+      end
+    end
+
+    context 'When is out of range' do
+      let!(:target) do
+        create(:target,
+               topic_id: topic_id)
+      end
+      it 'the number of macthers does not change' do
+        expect { subject }.not_to change { Match.count }
+      end
+    end
+
+    context 'When has the same user' do
```

Comment:
```suggestion
    context 'when has the same user' do
```
again: this `context` should be inside the `context` defined in line 57, it's also a case for when there are no matches